### DISCUSSION TITLE
RuntimeError: filter weights array has incorrect shape.

### DISCUSSION THREAD

**Discussion by JuergenTerpe** [colour-science/colour-demosaicing#57] [Q&A]
I'm a newbie in python programming, so maybe I'm doing something wrong. However, there are only limited parameters for these functions. I'm trying this: 

self.imageData = demosaicing_CFA_Bayer_bilinear(self.imageData, self.fits_header['BAYERPAT'])

self.imageData is a numpy array from a fits image (CFA RGGB) and it is in linear state, the numpy array is of type float32. I tried all functions and always get this error. I don't see any filter weights array to be passed as parameter. What is my mistake? 

Thanks in advance,
Juergen

**Comment by thomasmansencal** [comment]
Hi Juergen,

Is it possible to print the value of `self.fits_header['BAYERPAT']` and the shape of `self.imageData`?

Cheers,

Thomas

**Comment by JuergenTerpe** [comment]
Hi Thomas,

Sorry for the late answer. Sure it is possible: 

RGGB
(4168, 6224, 1)

It's a normal not debayered image from my astro camera, which shoots using RGGB bayer pattern and has a resolution of 4168x6224 pixels. I think the third shape value is the number of color values and this would be the expected value 1, because it's not debayered, so it's still a grayscale image. Please let me know which other information you require. It would be really helpful to have a chance to debayer these images within my tool I want to make public for the astrophotography community. 

Regards,
Jürgen

**Reply by thomasmansencal** [reply]
Ah!

This is something that we could/should handle better, try that:

```python
self.imageData = demosaicing_CFA_Bayer_bilinear(colour.utilities.tstack([self.imageData, self.imageData, self.imageData]), self.fits_header['BAYERPAT'])
```

The issue is that we are expecting a 3 channel image even though it is grayscale. I think that should fix your issue!

Cheers,

Thomas

**Comment by JuergenTerpe** [comment]
Thanks, Thomas. 

Unfortunately, it still crashes with the same error. 

  File "/Users/jt/PycharmProjects/ImageSessionAnalysis/FitsImage.py", line 76, in createDisplayImage
    self.imageData = demosaicing_CFA_Bayer_bilinear(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jt/PycharmProjects/ImageSessionAnalysis/venv/lib/python3.11/site-packages/colour_demosaicing/bayer/demosaicing/bilinear.py", line 130, in demosaicing_CFA_Bayer_bilinear
    R = convolve(CFA * R_m, H_RB)
        ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jt/PycharmProjects/ImageSessionAnalysis/venv/lib/python3.11/site-packages/scipy/ndimage/_filters.py", line 917, in convolve
    return _correlate_or_convolve(input, weights, output, mode, cval,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jt/PycharmProjects/ImageSessionAnalysis/venv/lib/python3.11/site-packages/scipy/ndimage/_filters.py", line 715, in _correlate_or_convolve
    raise RuntimeError('filter weights array has incorrect shape.')
RuntimeError: filter weights array has incorrect shape.

Might be something unexpected deeper inside. I can upload an image if you are interested. 

Regards,
Jürgen

**Reply by thomasmansencal** [reply]
Sorry, it is me being stupid, we want the opposite, i.e. no extra dim! Try the following:

```python
self.imageData = demosaicing_CFA_Bayer_bilinear(np.squeeze(self.imageData), self.fits_header['BAYERPAT'])
```

**Comment by JuergenTerpe** [comment]
Thank you, Thomas! 

Now it works!!

**Reply by thomasmansencal** [reply]
Awesome, I just updated the code in `develop` so that we squeeze systematically. Should avoid issues for future users!
